xc/device.ml

```diff
@@ -642,12 +642,37 @@ end
 
 module Vif = struct
 
+let write_extra_xenserver_keys ~xs device extra_xenserver_keys =
+	let domid = device.frontend.domid in
+	let devid = string_of_int device.frontend.devid in
+	let base_xenstore_path = Printf.sprintf "/local/domain/%d/xenserver/device/vif/%s" domid devid in
```

Comment:
This should be a function in device_common.ml, alongside `frontend_path_of_device`.